src/main/groovy/edu/oregonstate/mist/locations/resources/LocationResource.groovy

```diff
@@ -0,0 +1,42 @@
+package edu.oregonstate.mist.locations.resources
+
+import com.codahale.metrics.annotation.Timed
+import com.google.common.base.Optional
+import edu.oregonstate.mist.locations.core.Location
+import edu.oregonstate.mist.locations.db.LocationDAO
+import io.dropwizard.jersey.params.LongParam
+
+import javax.ws.rs.Consumes
+import javax.ws.rs.GET
+import javax.ws.rs.Path
+import javax.ws.rs.PathParam
+import javax.ws.rs.Produces
+import javax.ws.rs.WebApplicationException
+import javax.ws.rs.core.MediaType
+import javax.ws.rs.core.Response
+
+@Path("/locations")
+@Produces(MediaType.APPLICATION_JSON)
+@Consumes(MediaType.APPLICATION_JSON)
+class LocationResource {
+    private final LocationDAO locationDAO
+
+    //@todo: is this really needed?
+    LocationResource(LocationDAO locationDAO) {
+        this.locationDAO = locationDAO
+    }
+
+    @GET
+    @Timed
+    @Path('{id: \\d+}')
+    Location getById(@PathParam('id') LongParam id) {
+        final Location location = locationDAO.getLocationById(id.get())
+
+        if (!location) {
+            //@todo - does this also set the correct headers?
```

Comment:
I'm not sure how the WebApplicationException works but you might want to take a look at how I use the Javax Response type in my [dropwizard app](https://github.com/osu-mist/RestPR/blob/feature/RPR-3-Swagger-API-Specification/src/main/groovy/edu/oregonstate/mist/restpr/resources/UserResource.groovy#L70).

[Setting headers in the response build chaining is relatively easy too.](http://stackoverflow.com/questions/4621748/how-to-set-response-header-in-jax-rs-so-that-user-sees-download-popup-for-excel)